### PR TITLE
SliceUtf8.trim/leftTrim/rightTrim

### PR DESCRIPTION
**Pull request by haozhun**

### CODE REVIEW

File: src/main/java/io/airlift/slice/SliceUtf8.java [hunk 1]
```diff
@@ -309,6 +309,19 @@ public static Slice leftTrim(Slice utf8)
         return utf8.slice(position, length - position);
     }
 
+    /**
+     * Removes all white {@code whiteSpaceCodePoints} from the left side of the string.
```
Comment (dain) [inline src/main/java/io/airlift/slice/SliceUtf8.java]: typo, remove `white`

File: src/main/java/io/airlift/slice/SliceUtf8.java [hunk 2]
```diff
@@ -377,6 +428,44 @@ else if (minPosition <= position -4 && !isContinuationByte(utf8.getByte(position
         return position;
     }
 
+    private static int lastNonMatchPosition(Slice utf8, int minPosition, int[] codePointsToMatch)
```
Comment (dain) [inline src/main/java/io/airlift/slice/SliceUtf8.java]: add a comment mentioning this is an exact duplicate to both methods

File: src/test/java/io/airlift/slice/TestSliceUtf8.java [hunk 3]
```diff
@@ -85,6 +86,9 @@
                 .filter(codePoint -> getType(codePoint) != SURROGATE)
                 .toArray();
         STRING_ALL_CODE_POINTS = new String(ALL_CODE_POINTS, 0, ALL_CODE_POINTS.length);
+        WHITESPACE_CODE_POINTS = IntStream.rangeClosed(0, MAX_CODE_POINT)
```
Comment (dain) [inline src/test/java/io/airlift/slice/TestSliceUtf8.java]: Add one more simple test that uses non whitespace code points

**Comment by dain** [comment]
Looks good
